src/lib/features/frontend-api/frontend-api-controller.ts

```diff
@@ -229,11 +248,31 @@ export default class FrontendAPIController extends Controller {
             return;
         }
 
-        await this.services.frontendApiService.registerFrontendApiMetrics(
-            req.user,
-            req.body,
-            req.ip,
-        );
+        const environment =
+            await this.services.frontendApiService.registerFrontendApiMetrics(
+                req.user,
+                req.body,
+                req.ip,
+            );
+
+        if (
```

Comment:
I'd like to move this code to frontendApiService in the next PR. Added it here since it required minimal code changes to verify this behind a flag in production